src/extensions/default/AutoUpdate/node/AutoUpdateDomain.js

```diff
@@ -167,6 +167,96 @@
         }
     }
 
+    function parseInstallerLog(filepath, searchstring, encoding, callback) {
+        var line = "";
+        var str = "";
+        fs.stat(filepath)
+            .then(function (stats) {
+                if (stats) {
+                    fs.readFile(filepath)
+                        .then(function (chunkBuffer) {
+                            if (encoding) {
+                                str = chunkBuffer.slice(0, chunkBuffer.length).toString(encoding);
```

Comment:
@niteskum why are we not encoding the file content? Ideally, it should always be encoded, and thus, suitably handled.

Reply:
Since we are using promises you could avoid creating a [horizonal pyramids of doom ](http://blog.ministryofprogramming.com/prevent-pyramids-with-flat-promise-chains/) like shown in link.